src/d3a/d3a_core/global_objects.py

```diff
@@ -25,16 +25,34 @@ def __init__(self):
         self.area_stats_tree_dict = {}
         self.root_area = None
         self.external_tick_counter = None
+        self.current_feed_in_tariff = None
 
     def __call__(self, root_area, ticks_per_slot):
         self.root_area = root_area
         self.external_tick_counter = ExternalTickCounter(ticks_per_slot)
 
+    def _buffer_feed_in_tariff(self, area, current_market_slot):
+        """
+        This simplified recursion is sufficient as the infinite bus is expected to be in the
+        uppermost level of the tree
+        """
+        # the lazy is needed in order to avoid circular imports
```

Comment:
After converting this object to a singleton this should not be needed anymore. I have not checked the code, but mentally I foresee this object being imported by simulation.py and external strategies only, and I cannot visualise any circular import in these places.

Reply:
@spyrostz i exerienced this, that is why I added it like this
It is also the reason for changing the check for an external strategy to 
`if area.strategy is not None and area.external_connection_available`

Reply:
BTW; i merged 3333 into here already

Reply:
Interesting, then leave it as is